lib/elixir/lib/code.ex

```diff
@@ -661,8 +661,12 @@ defmodule Code do
     file = Keyword.get(opts, :file, "nofile")
     line = Keyword.get(opts, :line, 1)
 
-    with {:ok, tokens} <- :elixir.string_to_tokens(to_charlist(string), line, file, opts) do
-      :elixir.tokens_to_quoted(tokens, file, opts)
+    case :elixir.string_to_tokens(to_charlist(string), line, file, opts) do
```

Comment:
Just a note: the code before and after are equivalent. However, your version is clearer, so let's stick with it. :)